### PR TITLE
L3

### PR DESCRIPTION
**Pull request by zivmo99**

### CODE REVIEW

**Review by kapelnik — APPROVED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/input/jsonArch1PC.json [hunk 1]
```diff
@@ -28,12 +28,12 @@
     {
       "name": "w1",
       "id": 0,
-      "args": "[561,20,15,13,10,6]@0.001@[0,2,3,3,3,0]@6@0@561@6"
+      "args": "[3,2,1]@0.001@[0,2,0]@6@0@3@1"
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/input/jsonArch1PC.json]: Didn't we move to explicit parameters description? 
@6@3@1 means nothing to new user

File: src_cpp/cppBridge/NerlNIF.cpp [hunk 2]
```diff
@@ -428,6 +443,25 @@ static void* trainFun(void *arg){
     data_mat = Map<MatrixXd,0, Stride<Dynamic,Dynamic>>(trainPtr->data_label_mat.data(), trainPtr->rows, trainPtr->col,Stride<Dynamic,Dynamic>(1, FeaturesAndLabels));
     label_mat = Map<MatrixXd,0, Stride<Dynamic,Dynamic>>(&trainPtr->data_label_mat[trainPtr->col], trainPtr->rows, trainPtr->labels,Stride<Dynamic,Dynamic>(1, FeaturesAndLabels));
 
+    ofstream outdata; // outdata is to send the weights to a file
```
Comment (leondavi) [inline src_cpp/cppBridge/NerlNIF.cpp]: Why writing to a file within the NIF?

File: src_erl/erlBridge/nerlNetStatem.erl [hunk 3]
```diff
@@ -119,12 +116,14 @@ idle(cast, {predict}, State) ->
...
 
   io:fwrite("Set weights in wait state: \n"),
   
   %% Set weights TODO maybe send the results of the update
-  [WeightsList, BiasList, Biases_sizes_list, Wheights_sizes_list] = Ret_weights_tuple,
+  [WeightsList, BiasList, Biases_sizes_list, Wheights_sizes_list] = Ret_weights_list,
+
+  %% Make bias sizes and weights sizes as integer 
   NewBiases_sizes_list = [round(X)||X<-Biases_sizes_list],
   NewWheights_sizes_list = [round(X)||X<-Wheights_sizes_list],
```
Comment (leondavi) [inline src_erl/erlBridge/nerlNetStatem.erl:127]: Weights instead of wheights

**Review by leondavi — APPROVED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/input/jsonArch1PC.json (hunk 1 above)
Comment (zivmo99) [reply src_erl/Communication_Layer/http_Nerlserver/input/jsonArch1PC.json]: Yes. it's an old commit

File: src_cpp/cppBridge/NerlNIF.cpp (hunk 2 above)
Comment (zivmo99) [reply src_cpp/cppBridge/NerlNIF.cpp]: Added to debug now

**Pull request merged**